# sentor/test.py publishes these topics expr_1 and expr_2. 
# call services /pub_expr_1 and /set_expr_1 to start/stop publishing and set the topic arg, respectively. Same for expr_2.

- name : "/expr_1"
  signal_when:
    condition: "not published"
    timeout: 1.0
    safety_critical: True
    process_indices: [0]
    tags: ["expression 1"]
  signal_lambdas:
  - expression: "lambda msg : msg.data == False"
    timeout: 5.0
    safety_critical: True
    process_indices: [1]
    tags: ["expression 1"]
  execute:
  - log:
      message: "expression 1 is not published"
      level: "warn"
  - log:
      message: "expression 1 is False"
      level: "warn"
  default_notifications: False


- name : "/expr_2"
  signal_when:
    condition: "not published"
    timeout: 2.0
    safety_critical: True
    process_indices: [0]
    tags: ["expression 2"]
  signal_lambdas:
  - expression: "lambda msg : msg.data == False"
    timeout: 10.0
    safety_critical: True
    process_indices: [1]
    tags: ["expression 2"]
  execute:
  - log:
      message: "expression 2 is not published"
      level: "warn"
  - log:
      message: "expression 2 is False"
      level: "warn"
  default_notifications: False


- name: "/safe_operation"
  signal_lambdas:
    - expression: "lambda msg : msg.data == False"
      process_indices: [0]
    - expression: "lambda msg : msg.data == True"
      process_indices: [1]
  execute:
  - log:
      message: "Not safe to operate"
      level: warn
  - log:
      message: "Safe to operate"
      level: info
  timeout: 0.1
  default_notifications: False

